### PR TITLE
More new features

### PR DESCRIPTION
**Pull request by erwald**
Add more new features, and move some preprocessing to the Makefile.

    You now need to run "make processdata" before going through the
    notebook, and unfortunately it takes a very long time, but at least
    we only need to do it once ...

    I tried to aggregate everything from the transactions data frames
    in a sensible way, but I may have missed something, so extra eyes
    would be appreciated ...

### CODE REVIEW

**Comment by erwald** [comment]
> ```
> jkan@fl041-5:kaggle-competition-elo-loyalty-prediction (more-new-features)$ make processdata
> mkdir data/processed
> mkdir: data/processed: File exists
> make: *** [data/processed] Error 1
> ```
> It's a good idea to use `mkdir -p` so that `mkdir` doesn't fail when the directory already exists. (`mkdir -p` is also used to create directory hierarchies in one go. I have no idea why both functionalities are bundled up in the same option.)

Aha, that's better, yes -- fixed!

**Comment by erwald** [comment]
> It almost looks like we're already overfitting even though the result still a bit off perfect.
> Wouldn't that indicate that we still need better feature engineering/understanding?

We are overfitting, and we need better feature engineering, but I am not sure the two are directly related in that way?

We could combat the overfitting problem by increasing dropout and/or weight decay.
